src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -516,44 +533,415 @@ abstract class ManagedStrategy(
         return 0
     }
 
-    /**
-     * This method is executed before a shared variable read operation.
-     * @param iThread the number of the executed thread according to the [scenario][ExecutionScenario].
-     * @param codeLocation the byte-code location identifier of this operation.
-     */
-    internal fun beforeSharedVariableRead(iThread: Int, codeLocation: Int, tracePoint: ReadTracePoint?) {
+    override fun lock(monitor: Any, codeLocation: Int): Unit = runInIgnoredSection {
+        val tracePoint = if (collectTrace) {
+            val iThread = currentThread
+            MonitorEnterTracePoint(
+                iThread = iThread,
+                actorId = currentActorId[iThread],
+                callStackTrace = callStackTrace[iThread],
+                stackTraceElement = CodeLocations.stackTrace(codeLocation)
+            )
+        } else {
+            null
+        }
+        beforeLockAcquire(codeLocation, tracePoint, monitor)
```

Comment:
Please raise an issue for the `ManagedStrategy` class refactoring to avoid the logic separation and using the legacy API here.